internal/identify/identify.go

```diff
@@ -62,63 +65,109 @@ type scrapeResult struct {
 	source ScraperSource
 }
 
-func (t *SceneIdentifier) scrapeScene(ctx context.Context, scene *models.Scene) (*scrapeResult, error) {
+func (t *SceneIdentifier) scrapeScene(ctx context.Context, txnManager txn.Manager, scene *models.Scene) (*scrapeResult, error) {
 	// iterate through the input sources
 	for _, source := range t.Sources {
 		// scrape using the source
-		scraped, err := source.Scraper.ScrapeScene(ctx, scene.ID)
+		results, err := source.Scraper.ScrapeScenes(ctx, scene.ID)
 		if err != nil {
 			logger.Errorf("error scraping from %v: %v", source.Scraper, err)
 			continue
 		}
 
-		// if results were found then return
-		if scraped != nil {
-			return &scrapeResult{
-				result: scraped,
-				source: source,
-			}, nil
+		if len(results) > 0 {
+			options := t.getOptions(&source)
+			if len(results) > 1 && options.SkipMultipleMatches != nil && *options.SkipMultipleMatches {
+				if options.SkipMultipleMatchTag != nil && len(*options.SkipMultipleMatchTag) > 0 {
+					// Tag it with the multiple results tag and ignore
+					err := t.addTagToScene(ctx, txnManager, scene, options.SkipMultipleMatchTag)
+					if err != nil {
+						return nil, err
+					}
+					return nil, nil
+				}
```

Comment:
I think this violates the intention of the `scrapeScene` method, which should only be concerned with getting and returning the results of the scrape. Instead, if multiple results are found and the options indicate that multiple matches should not be returned, then it should return an error value instead. This should be checked in the calling code, which can either integrate the multiple match tag in the `modifyScene` code, or just do the `addTagToScene` in a separate if branch.